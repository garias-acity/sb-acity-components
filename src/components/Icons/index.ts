export * from './HeartIcon'
export * from './ChevronLeft'
export * from './ChevronRight'