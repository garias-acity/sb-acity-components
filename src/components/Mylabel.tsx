import './Mylabel.css'

interface Props {
    /**
     * Text to display
     */
    label: string;
    /**
     * Label size
     */
    size?: 'normal' | 'h1' | 'h2' | 'h3';
    allCaps?: boolean;
    color?: 'primary' | 'secondary' | 'tertiary';
    fontColor?: string;
}

/**
 * AllCaps boolean
 * color: text-primary... 
 * font-color?: string
 */

export const Mylabel = ({ label, size = 'normal', allCaps = false, color = 'primary', fontColor= '#000' }: Props) => {
  return (
    <div className={ `${size} text-${color} label` } style={{ color: fontColor }}>
        {
            allCaps ? label.toLocaleUpperCase() : label
        }
    </div>
  )
}