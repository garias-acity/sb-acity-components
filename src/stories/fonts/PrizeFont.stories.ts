import { Meta, StoryObj } from "@storybook/react";
import { Coin, PrizeFont } from "@/components/fonts";

const meta: Meta<typeof PrizeFont> = {
  title: "Fonts/PrizeFont",
  component: PrizeFont,
  tags: ["autodocs"],
  parameters: {
    layout: "centered",
  },
  args: {
    amount: "X,XXX",
    coin: Coin.PEN,
  },
  argTypes: {
    coin: {
      description: `${Coin.PEN} | ${Coin.USD}`,
      options: [Coin.PEN, Coin.USD],
      control: { type: "select" },
    },
  },
};

export default meta;

type Story = StoryObj<typeof PrizeFont>;

export const Example: Story = {
  args: {
    amount: "X,XXX",
    coin: Coin.PEN,
  },
};
export const Thousand: Story = {
  args: {
    coin: Coin.PEN,
    amount: "1,000",
  },
};
