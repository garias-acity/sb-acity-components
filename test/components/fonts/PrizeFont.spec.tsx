import { PrizeFont } from "@/components/fonts";
import { expect, suite, test } from "vitest";
import { render } from "@testing-library/react";

suite("PrizeFont", () => {
  test("Renders snapshot", () => {
    expect(<PrizeFont amount="10,000" />).toMatchSnapshot();
  });

  test("Renders default currency sign", () => {
    const rendered = render(<PrizeFont amount="10,000" />);
    const content = rendered.getByLabelText("content").innerHTML;
    expect(content).toContain("S/");
  });
});
