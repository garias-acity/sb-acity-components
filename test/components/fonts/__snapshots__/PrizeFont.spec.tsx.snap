// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrizeFont > Renders snapshot 1`] = `
<Memo
  amount="10,000"
/>
`;
